### ISSUE TITLE
glitter for named graphs

### ISSUE THREAD

**Issue by nleguillarme** [lvaudor/glitter#98]
Hi,

I am wondering if glitter can be used to build queries 

1. that specify the dataset to be used for matching using the FROM and FROM NAMED clauses
2. that use the GRAPH keyword to match patterns against named graphs

If not, it would be great to see these get included in the package.

**Comment by lvaudor** [comment]
Hi!
Thanks for your feedback. Hmm let us have a think here with @maelle... We have not worked with named graphs yet I guess we need practical examples to get a grasp of how we could use them... A few explanations can be found here maybe: [](https://levelup.gitconnected.com/working-with-rdf-database-named-graphs-a5ddab447e91)

**Comment by lvaudor** [comment]
@nleguillarme would you be able to provide a (SPARQL) query example that you'd be trying to reproduce here?

**Comment by nleguillarme** [comment]
Hi @lvaudor 
For instance, if I look for observation records of some traits for species "Armillaria gallica" in my triplestore, and want the iri of the named graph containing the observation triples, I would write the following query:

```
SELECT DISTINCT ?observation ?g
WHERE {
  VALUES ?sciName {"Armillaria gallica"}
  ?taxon rdfs:label ?sciName.
  GRAPH ?g {
    ?observation obo:OBI_0000293 ?organism.
    ?organism sesame:directType ?taxon.
  }
}
```

**Comment by lvaudor** [comment]
Thanks! So, it's a local triplestore? I guess, if so, you'd also have a question along the lines of the Issue [support for additional backends? #10 ](https://github.com/lvaudor/glitter/issues/10)?

**Comment by nleguillarme** [comment]
Exactly, it's a private instance of GraphDB we set up one one of our servers.
I use glitter to encapsulate SPARQL queries in R functions, as a user-friendly interface to our triplestore.

**Comment by lvaudor** [comment]
Great! Thanks for the feedback @nleguillarme  We'll see what we can do about searching specific graphs. For now we do not use private triplestores and so had not run yet into the question of sub-graphs so far... Am I right guessing that each of your graphs would correspond to specific geographic sites?

**Comment by maelle** [comment]
@lvaudor what glitter interface would you envision for this?

**Comment by maelle** [comment]
related: #144 (FROM)
